# Derivation of average rate of return of S&P 500 (1976-2016)
---

`r = average rate of return`

`n = number of years elapsed`

`price_start * (1 + r)^n = price_end`

`(1 + r)^n = price_end / price_start`

`1 + r = (price_end / price_start)^(1/n)`

`r = (price_end / price_start)^(1/n) - 1`

In [31]:
t_start = 1976
t_end = 2016
n = t_end - t_start
price_start = 103.79  # price of S&P500 at end of August 6, 1976
price_end = 2173.60   # price of S&P500 at end of July 29, 2016

r = (price_end / price_start)**(1.0/n) - 1
r_in_percent = r * 100
print "average rate of return of market gain: %.2f%%" % r_in_percent

d_start = 16.77
d_end = 44.46
dr = (d_end / d_start)**(1.0/n) - 1
dr_in_percent = dr * 100
print "averate rate of return of dividends: %.2f%%" % dr_in_percent

average rate of return of market gain: 7.90%
averate rate of return of dividends: 2.47%


# How to derive the average rate of return
The above calculation shows that **the average rate of return of the S&P500 from 1976-2016 was 7.90%**. We're interested in the S&P500's average rate of return because that's the rate we need to beat if we're going to engage in stockpicking to beat the market. The S&P500 doesn't represent the entire market, but it's a good proxy. (If you check the [NYSE](http://www.nasdaq.com/screening/companies-by-industry.aspx?exchange=NYSE) and the [NASDAQ](http://www.nasdaq.com/screening/companies-by-industry.aspx?exchange=NASDAQ), you'll see that there are about 6000 publicly listed stocks in the two largest U.S. exchanges. While the S&P500 contains about one-twelfth the number of public listed companies, it's a good proxy for the total U.S. stock market.) Some stocks also return **dividends**, and in our case, we calculated the rate to be **2.47%**.

Two things eat into your stock market gains: taxes and inflation. Assuming we put our money in the stock market and never sell to realize gains (or losses), and assuming we automatically reinvest all dividends into purchasing more shares, we can avoid taxes. That leaves us with inflation. This means our **10.37%** total average rate of return isn't the final rate of return. Using the [US inflation calculator](http://www.usinflationcalculator.com/), a dollar in 1976 is worth \$4.24 in 2016, for a cumulative rate of inflation of 323.6%. Let's calculate the average rate of inflation from these numbers.

In [42]:
cumulative_rate_of_inflation = 4.236
average_rate_of_inflation = cumulative_rate_of_inflation ** (1.0/(2016-1976)) - 1
print "average rate of inflation: %.2f%%" % (average_rate_of_inflation * 100)

average rate of inflation: 3.67%
average rate of inflation: 2.13%


# How to derive the average rate of inflation

The above calculation shows that the average rate of inflation over 40 years is about **3%**. This agrees with published numbers from [Ben Graham's The Intelligent Investor](https://www.amazon.com/Intelligent-Investor-Definitive-Investing-Essentials/dp/0060555661) and other common investing tutorials.

Note that the cumulative rate of inflation was 323.6%. To get this number, see that \$1 is worth \$4.24, so that means the **rate** is *4.24 - 1*. The percentage is simply the rate multiplied by 100.

Also note that the reverse calculation is done to obtain the average rate of inflation.

# Real rate of return

With the inflation rate at 3% and the average rate of return of the S&P500 at 10.37%, that means real gains from investing in a total U.S. stock market index fund is 7.37% (again, assuming no taxes). 